versions.yaml

```diff
@@ -284,6 +284,22 @@ externals:
     url: "https://github.com/containerd/nydus-snapshotter"
     version: "v0.3.3"
 
+  open-policy-agent:
+    description: "Open Policy Agent"
+    url: "https://github.com/open-policy-agent/opa"
+    version: "v0.55.0"
+    meta:
+      # - If an OPA package is available for the Guest image distro, that package
+      #   is used instead of the binary below.
+      #
+      # - TODO: if an OPA package is not available for the Guest image distro,
+      #   Kata should cache the OPA source code, toolchain information, etc.
+      #   OPA should be built from the cached source code instead of downloading
+      #   this binary.
+      #
+      # yamllint disable-line rule:line-length
+      binary: "https://github.com/open-policy-agent/opa/releases/download/v0.55.0/opa_linux_amd64_static"
```

Comment:
So, it's nice that OPA has a binary for amd64 / arm64, but unfortunately it doesn't provide the same for other architectures that we're using (s390x, and maybe powerpc64).

Instead of adding the `binary` entry here, I'd recommend:
* Having a generic link `tarball_url` / `binary_url` which would point https://github.com/open-policy-agent/opa/releases/download/
  * This is similar to: https://github.com/kata-containers/kata-containers/blob/main/versions.yaml#L221C3-L221C78
* Have a script under https://github.com/kata-containers/kata-containers/tree/main/tools/packaging/static-build, where we'd download the binary if we're building for x86_64 / arm64, build the binary if we're on s390x / powerpc64
  * Something similar is done here: https://github.com/kata-containers/kata-containers/blob/main/tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh
  * The build / download should be done inside a container
* Call the script, which always will give you the binary (or error out in case of build fails), and then we can use it
  * This will make things way friendlier for different arches.

Reply:
@BbolroC, @Amulyam24, can you help with the s390x / powerpc build side of the things?

Reply:
Yes, I can, but is it okay to go over this next Tuesday?

Reply:
Sure @fidencio, I can work on this next week. Will coordinate with @BbolroC if needed :)

Reply:
FYI: I was able to generate a static binary `opa_linux_s390x_static` on a `amd64` machine with the following:

```
$ go version
go version go1.21.0 linux/amd64
$ git clone -b v0.55.0 https://github.com/open-policy-agent/opa.git
$ cd opa
$ export GOARCH=s390x
$ make clean generate
$ make ci-go-ci-build-linux-static
$ $ tree --du -h _release
_release
└── [ 39M]  0.55.0
    ├── [ 39M]  opa_linux_s390x_static
    └── [  89]  opa_linux_s390x_static.sha256
$ file _release/0.55.0/opa_linux_s390x_static
_release/0.55.0/opa_linux_s390x_static: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), statically linked, Go BuildID=lgngaupMJCkzniJiPQae/zxxUQm2pjE0oRsFgizYe/dCaqFReGFIt0ZDCUd3xO/rLftgYdjD6B86PJjIXAR, with debug_info, not stripped
```

Reply:
Thanks everyone! New issue to tracked this work: https://github.com/kata-containers/kata-containers/issues/7615